src/copaw/agents/tools/browser_control.py

```
@@ -504,26 +589,81 @@ def on_page(page):
+            if extra_args:
+                launch_kwargs["args"] = extra_args
+            pw_browser = await pw.chromium.launch(**launch_kwargs)
         context = await pw_browser.new_context()
         _attach_context_listeners(context)
         _state["playwright"] = pw
         _state["browser"] = pw_browser
         _state["context"] = context
+        _state["_last_browser_error"] = None
+        _touch_activity()
+        _start_idle_watchdog()
         return True
```

Ruff (0.15.2)

[warning] Consider moving this statement to an `else` block

(TRY300)